cli/internal/state/state.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+// package state defines the structure of the Constellation state file.
+package state
+
+// State describe the entire state to describe a Constellation cluster.
+type State struct {
+	Version        string `yaml:"version"`
+	Infrastructure `yaml:"infrastructure"`
+}
+
+// Infrastructure describe the state related to the cloud resources of the cluster.
+type Infrastructure struct {
+	UID               string   `yaml:"uid"`
+	PublicIP          string   `yaml:"publicIP"`
+	InitSecret        string   `yaml:"initSecret"`
+	APIServerCertSANs []string `yaml:"apiServerCertSANs"`
+	Azure             *Azure   `yaml:"azure"`
+	GCP               *GCP     `yaml:"gcp"`
```

Comment:
It seems more idiomatic to me to make this an interface and add marshalling methods. Would prevent setting both keys, makes nil checks obsolete, makes the intent of the property clearer. Was there a specific reason you chose this approach, apart from the fact that this is how we do it in the config?

Reply:
you mean to make this field contain a CSP interface similar to the attestation config?
In the end the properties of the CSPs are very different so I don't see how to generalize the fields in an interface.
What is unclear about the intent of the property?

Do you have a suggestion?

Reply:
True. It isn't clear how to write that interface.